src/content/docs/en/guides/incremental-static-regeneration.mdx

```diff
@@ -0,0 +1,272 @@
+---
+title: Incremental Static Regeneration
+---
+If your website fetches content from a CMS or a database that updates infrequently, you can use a technique called **Incremental Static Regeneration**, or ISR, to update your site without rebuilding the entire project. This is a great way to keep your site up-to-date while still enjoying the benefits of static hosting.
+
+:::note[SSG vs SSR vs ISR]
+ISR is often presented as an alternative to Static Site Generation (SSG) and Server Side Rendering (SSR), but it is more helpful to think of it as SSR with caching.
+:::
+
+:::note
+Incremental static regeneration is different from **incremental builds**, which speeds up full rebuilds by reusing work done by the previous builds.
+:::
+
+## Why ISR?
+- Your site has a lot of pages (more than one thousand). Build times can be shortened when using SSR or ISR.
+- Your content changes infrequently (less frequently than once in 5 minutes on average).
+- Your site gets a lot of visitors. It can be wasteful to render the same page for every visitor, and you can avoid it using ISR.
+- The CMS or database you are using is slow, or charges by request. With ISR, the requests to your backend will become much less frequent.
+
+## Why not ISR?
+- Your content is located inside your project (for example, as markdown files in `src/content`).
+- You don't want the hassle of running a server or using a serverless platform.
+- Your website offers logged-in experiences. If each user needs to be sent a slightly different HTML from others, caching the page may not be possible.
+- The structure of your site (layouts, and styling) changes frequently. Changes to the structure require a full rebuild.
+
+## Implementing ISR for Your Project
+
+ISR involves saving the response such that it can be reused for future requests. The exact implementation depends on where you are deploying.
+
+If you aren't using an adapter already, you are going to need to add one. See the [integrations guide](/en/guides/integrations-guide/) for specific instructions. In your config, make sure that the `output` is either `"server"` or `"hybrid"`. Note that when using "hybrid", the pages you want generated, cached, and invalidated will need to be opted-out of prerendering. Otherwise, they will only be rendered once during build. See the [server side rendering guide](/en/guides/server-side-rendering/#opting-out-of-pre-rendering) to see how.

```

Comment:
```suggestion
If you aren't using an adapter already, you are going to need to add one. See the [integrations guide](/en/guides/integrations-guide/) for specific instructions. In your config, make sure that the `output` is either `"server"` or `"hybrid"`. Note that when using "hybrid", the pages you want generated, cached, and invalidated will need to be opted-out of prerendering. Otherwise, they will only be rendered once during build. See the [server side rendering guide](/en/guides/server-side-rendering/#opting-out-of-pre-rendering-in-hybrid-mode) to see how.

```